cassovary-core/src/main/scala/com/twitter/cassovary/util/io/AdjacencyListGraphReader.scala

```diff
@@ -111,10 +110,7 @@ class AdjacencyListGraphReader[T] (
             i += 1
           }
 
-          holder.id = internalNodeId
-          holder.edges = outEdgesArr
-          holder.maxId = newMaxId
-          holder
+          NodeIdEdgesMaxId(internalNodeId, outEdgesArr, newMaxId)
```

Comment:
For the case of immutable graph we would prefer to avoid creating the object here, don't we @pankajgupta ?